source/postprocess/dynamic_topography.cc

```diff
@@ -110,9 +117,14 @@ namespace aspect
               // for each of the quadrature points, evaluate the
               // stress and compute the component in direction of the
               // gravity vector
+              
+              double dynamic_topography_x_volume = 0;
+              double volume = 0;
+              Point<dim> location;
+
               for (unsigned int q=0; q<quadrature_formula.size(); ++q)
                 {
-                  const Point<dim> location = fe_values.quadrature_point(q);
+                  location = fe_values.quadrature_point(q);
```

Comment:
The way you reorganize the code here is slightly confusing. You introduce the variable `location` at the top, before the loop, then you set it in each iteration of the loop, and then you use the value after the loop. This can only work if the loop only goes over a single quadrature point -- which really is the case here, but that's a different point.

How about this: add a comment before the loop of the kind
```
  Compute the integral of the dynamic topography function over the entire cell, by looping over all quadrature points (currently, there is only one, but the code is generic).
```
Then move the declaration of the `location` variable into the loop. Further down below, you then push back the averaged value not by re-using the `location` variable, but by explicitly using `cell->center()` as the location. This way it's clear that you're associating the value not with some arbitrary quadrature point (that happens to be the last point you visited), but a purposefully chosen point.